sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -181,11 +182,7 @@ abstract class AggregationIterator(
         // Process all expression-based aggregate functions.
         updateProjection.target(currentBuffer)(joinedRow(currentBuffer, row))
         // Process all imperative aggregate functions.
-        var i = 0
-        while (i < updateFunctions.length) {
-          updateFunctions(i)(currentBuffer, row)
-          i += 1
-        }
+        updateFunctions.foreach(updateFunction => updateFunction(currentBuffer, row))
```

Comment:
Ah, right, i rushed both your comment and blog (please excuse this as it's my phone and I am outside). Yeap. I am fine with this.